first pass on new backend service client

marenas opened the pull request:
first pass on new backend service client

Commit:
first pass on new backend service

yonpols commented on RouteToDo/BackendRepository.h:
```diff
@@ -33,6 +33,7 @@
 #pragma mark - Actions
 
 - (void)toggleRouteFavoriteWithUser:(User *)user route:(Route *)route completion:(void (^)(NSError *error))completion;
+- (void)untoggleRouteFavoriteWithUser:(User *)user route:(Route *)route completion:(void (^)(NSError *error))completion;
```
I don't think this is necessary. Toggle means its going to flip the value to true or false

marenas replied:
makes sense. fixed.

yonpols commented:
Can we remove parse from the pod file?

marenas commented:
will remove parse as well.

Commit:
- removed parse
- removed untoggle function
- moved http repo as main backend repo protocol
- commented out many function until we implement them in the backend service. i will update them as we update the backend service.
